File: src/main/java/com/telmomenezes/synthetic/cli/RandGen.java
package com.telmomenezes.synthetic.cli;

import com.telmomenezes.synthetic.NetParams;
import com.telmomenezes.synthetic.generators.Generator;
import com.telmomenezes.synthetic.generators.GeneratorFactory;

public class RandGen extends Command {
    
	@Override
	public String name() {return "randgen";}
	
	
	@Override
	public String help() {
		String help = "Create a random generator program.\n";
		help += "$ synt randgen -oprg <out_generator>\n";
		help += "-undir if network is undirected.\n";
		return help;
    }
	
	
	@Override
	public boolean run() throws SynCliException {
    	String outProg = getStringParam("oprg");
    	boolean directed = !paramExists("undir");
        
    	NetParams netParams = new NetParams(0, 0, directed,false);
        Generator gen = GeneratorFactory.create("exo", netParams, 0);
		if (gen == null) {
			System.err.println("could not load program.");
			return false;
		}
        gen.initRandom();
        
        gen.getProg().write(outProg);
    	
        System.out.println("done.");
        
        return true;
    }
}

File: synthetic/commands/rand_gen.py
from synthetic.generator import create_generator
from synthetic.commands.command import *


class RandGen(Command):
    def __init__(self, cli_name):
        Command.__init__(self, cli_name)
        self.name = 'rand_gen'
        self.description = 'create a random generator program'
        self.mandatory_args = ['oprg']
        self.optional_args = ['undir', 'gentype']

    def run(self, args):
        self.error_msg = None

        out_prog = args['oprg']

        directed = not args['undir']
        gentype = arg_with_default(args, 'gentype', DEFAULT_GEN_TYPE)

        gen = create_generator(directed, gentype, init_random=True)
        gen.prog.write(out_prog)

        print('done.')

        return True
